charts/opentelemetry-kube-stack/templates/_config.tpl

```diff
@@ -227,16 +227,23 @@ receivers:
 {{- end }}
 
 {{- define "opentelemetry-kube-stack.collector.applyClusterMetricsConfig" -}}
-{{- $config := mustMergeOverwrite (include "opentelemetry-kube-stack.collector.clusterMetricsConfig" .collector | fromYaml) .collector.config }}
+{{- $config := mustMergeOverwrite (include "opentelemetry-kube-stack.collector.clusterMetricsConfig" (dict "collector" .collector "namespace" .namespace) | fromYaml) .collector.config }}
 {{- if and (dig "service" "pipelines" "metrics" false $config) (not (has "k8s_cluster" (dig "service" "pipelines" "metrics" "receivers" list $config))) }}
 {{- $_ := set $config.service.pipelines.metrics "receivers" (append ($config.service.pipelines.metrics.receivers | default list) "k8s_cluster" | uniq)  }}
+{{- $_ := set $config.service "extensions" (append ($config.service.extensions | default list) "k8s_leader_elector/k8s_cluster" | uniq)  }}
 {{- end }}
 {{- $config | toYaml }}
 {{- end }}
 
 {{- define "opentelemetry-kube-stack.collector.clusterMetricsConfig" -}}
+extensions:
+  k8s_leader_elector/k8s_cluster:
+    auth_type: serviceAccount
+    lease_name: k8s.cluster.receiver.opentelemetry.io
+    lease_namespace: {{ .namespace }}
```

Comment:
I know it will merge cleanly, but if the leader elector extension ever needs changed we'll have to make sure to update all the instances. Can you move this to its own template?

Reply:
yeah, nice idea! Tomorrow I'll do it and I'll rebase my branch too

Reply:
PTAL @TylerHelmuth , I've extracted the generation to another block and reused it for both cases